### ISSUE TITLE
Cisco AireOS WLC 2504 not Working

### ISSUE THREAD

**Issue by Chris-B-Chicken** [ytti/oxidized#1012]
Hello,
the AireOS for a Cisco WLC 2504 Version 8.3.102 is not working properly. It won't connect to the Controller. Normal SSH out of the Docker-Host work. 

root@docsuk:/etc/oxidized# ssh 192.168.0.51

(Cisco Controller)
User: oxidized
Password:************
(Cisco Controller) >

Oxidized Bash looks like this:
I, [2017-09-04T09:54:48.189964 #1]  INFO -- : Configuration updated for AZB/skcsw-azb09                                                                         
W, [2017-09-04T09:55:16.147806 #1]  WARN -- : 192.168.0.51 raised Timeout::Error with msg "execution expired"                                                   
W, [2017-09-04T09:55:16.197512 #1]  WARN -- : skcwc-stby1 status no_connection, retry attempt 1                                                                 
W, [2017-09-04T09:55:41.338590 #1]  WARN -- : 192.168.0.51 raised Oxidized::PromptUndetect with msg "                                                           
 not matching configured prompt (?-mix:^\([^\)]+\)\s>)"                                                                                                         
W, [2017-09-04T09:55:42.204235 #1]  WARN -- : skcwc-stby1 status no_connection, retry attempt 2                                                                 
W, [2017-09-04T09:56:06.617421 #1]  WARN -- : 192.168.0.51 raised Oxidized::PromptUndetect with msg "                                                           
 not matching configured prompt (?-mix:^\([^\)]+\)\s>)"                                                                                                         
W, [2017-09-04T09:56:07.210608 #1]  WARN -- : skcwc-stby1 status no_connection, retry attempt 3                                                                 
W, [2017-09-04T09:56:35.022536 #1]  WARN -- : 192.168.0.51 raised Timeout::Error with msg "execution expired"                                                   
W, [2017-09-04T09:56:35.217615 #1]  WARN -- : skcwc-stby1 status no_connection, retries exhausted, giving up 
´
Source-Map:
map:
      name: 0
      ip: 1
      model: 2
      group: 3
      username: 4
      password: 5

router.db:
skcwc-stby1:192.168.0.51:aireos:STR:oxidized:somePasswordHere

Im okay with linux, but bad in ruby programming. I don't get the syntax out of it... can you help me? 
Do you need more info?

**Comment by laf** [comment]
Is `(Cisco Controller)` the prompt? If so that doesn't match: `^\([^\)]+\)\s>` so it would need updating.

**Comment by Chris-B-Chicken** [comment]
When i connect to the controller via ssh first entry is 
(Cisco Controller)
User: _
So than i can enter a username. The configuration prompt is:
(Cisco Controller) > show run-config (for example)
Where are the *.rb files located in a docker-environment?

**Comment by laf** [comment]
So that should match fine but the debug output shows it's not matching so I would try dropping the > from prompt detection maybe to test?

**Comment by Chris-B-Chicken** [comment]
Ye i wanna test it, but i cant rewrite the aireos.rb because i cant find it in my docker container. i dont know where it is located. Do you got any suggestions?

**Comment by laf** [comment]
You can place custom models in "~/config/oxidized/model/" for them to be picked up for testing, same filenames as the core models will override those so just pop it in that dir.

**Comment by Chris-B-Chicken** [comment]
Nope, it wont work. Mabye it is interesting that the shell need about 5seconds to get a connection?

**Comment by laf** [comment]
No docs on the models I'm afraid, only looking what is already done and then checking what those lines of code do in Ruby.

**Comment by ksedge** [comment]
I am having the same issue with the Cisco WLC.

I am also a bit new to Oxidized and ruby, so I suspect if I knew what I was doing, I could do this myself

I tried dropping the > from the prompt as suggested and then it logs in but doesn't run any commands.

I get this now via debug; 


(Cisco Controller)
User: admin
Password:*******
(Cisco Controller) >

Then it disconnects and runs again

I am trying to work out what triggers it to run the commands to get the config? is it still this same regex that I need to make match both with and without a ">"?

**Comment by ksedge** [comment]
Ignore that - for some reason it has started working now!

**Comment by JB-BR** [comment]
Hello, 

I found the root cause. 
A user with ReadOnly rights cannot execute : "config paging disable"

`(Cisco Controller) >config paging disable
Incorrect usage.  Use the '?' or <TAB> key to list commands.`

Then, when executing "show run-config commands" oxidized is blocked by the "more" prompt and gives up : 

`
(Cisco Controller) >show run-config commands
 802.11a 11nSupport a-mpdu tx scheduler enable

**--More-- or (q)uit**
`

**Solution** : give oxidized ReadWrite rights on the WLC

**Comment by jsynack** [comment]
@JB-BR - Thanks for the information/debugging!

Is there any chance that you would be able to supply any configuration information/examples in the form of a pull request to the project?

https://github.com/ytti/oxidized/tree/master/docs/Model-Notes - is where you would want to create the information / implementation notes.

This could be quite beneficial to other users that encounter the same item.

**Comment by laf** [comment]
@jsynack The solution is to give the user logging in write access to run the config paging disable command.

Without that it would need detection in place for `more` and allowing the config backup to continue.

**Comment by jsynack** [comment]
@laf - yes - I got that from the comments - was just trying to get config snippets added for the Model-specific notes - since I don't have any of the WLCs to write anything up.

**Comment by tychoo** [comment]
I've validated the user account being used in my instance does have permissions to run config paging disable, but still get the following:

User: not matching configured prompt (?-mix:^\([^\)]+\)\s>)"

**Comment by laf** [comment]
@jsynack Can you confirm if the prompt detection needs updating as well or does the model work if the user is set correctly?

**Comment by sousavarela** [comment]
Hi,
I have the same problem, has anyone solved it? or can we work together to resolve?
I'm using the standard aireos.rb file

```
10.0.0.0 raised Net::OpenTimeout (rescued Timeout::Error) with msg "t>
WARN -- : wlc/10.0.0.0 status no_connection, retry attempt 1
WARN -- : 10.0.0.0 raised Net::OpenTimeout (rescued Timeout::Error) with msg "t>
```
Thanks
